### PR TITLE
[discord-game-sdk] use updated domain name

### PR DESCRIPTION
**Pull request by J-Human**
**Describe the pull request**
This updates the external URLs to use the base domain name of `discord` instead of `discordapp` since [the company changed the name last year](https://support.discord.com/hc/en-us/articles/360042987951-Discordapp-com-is-now-Discord-com).
- #### What does your PR fix?  
N/A

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
As per the configuration file, all 3 are supported. No, I didn't.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
Done.

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**
N/A

### CODE REVIEW

File: ports/discord-game-sdk/vcpkg.json [hunk 1]
```diff
@@ -3,6 +3,6 @@
   "version-string": "2.5.6",
   "port-version": 2,
```
Comment (NancyLi1013) [inline ports/discord-game-sdk/vcpkg.json:4]:
```suggestion
  "version": "2.5.6",
  "port-version": 3,
```
This might be `version `here.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Please run `vcpkg x-add-version discord-game-sdk` to update version files after you address the review suggestions.

File: ports/discord-game-sdk/vcpkg.json (hunk 1 above)
Comment (J-Human) [reply ports/discord-game-sdk/vcpkg.json]: Addressed.

**Comment by J-Human** [comment]
> Please run `vcpkg x-add-version discord-game-sdk` to update version files after you address the review suggestions.

Done, it has been committed.

File: versions/d-/discord-game-sdk.json [hunk 2]
```diff
@@ -1,7 +1,12 @@
 {
   "versions": [
     {
-      "git-tree": "17cdcbe9e20da2ed085fc2de181ae426d62f2609",
+      "git-tree": "4ddd26248ed26eb2676c96330e8ae5a0c3f82a3d",
+      "version": "2.5.6",
+      "port-version": 3
+    },
+    {
+      "git-tree": "1e36656365b603448365695d858cf2eaf3aff0bd",
```
Comment (NancyLi1013) [inline versions/d-/discord-game-sdk.json:9]: This is not correct. We should not change the content in the last version. Adding a new item for the new version changes is right action.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: versions/d-/discord-game-sdk.json (hunk 2 above)
Comment (J-Human) [reply versions/d-/discord-game-sdk.json:14]: I think I fixed it, please check. Thanks!

File: versions/d-/discord-game-sdk.json [hunk 3]
```diff
@@ -1,7 +1,17 @@
...
+      "version": "2.5.6",
+      "port-version": 3
+    },
     {
       "git-tree": "17cdcbe9e20da2ed085fc2de181ae426d62f2609",
+      "version": "2.5.6",
+      "port-version": 3
+    },
+    {
+      "git-tree": "1e36656365b603448365695d858cf2eaf3aff0bd",
```
Comment (NancyLi1013) [inline versions/d-/discord-game-sdk.json:14]:
```suggestion
```
This should be removed. You add this twice. Every PR only has one item for the changes.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Looks good, thank you!

**Pull request merged**